.github/workflows/gcp_build_and_deploy.yml

```diff
@@ -0,0 +1,181 @@
+name: Build and deploy Go clean architecture application
+
+on:
+  push:
+    branches:
+      - BAC-*
+      - main
+      - develop
+    tags:
+      - v*
+
+env:
+  GITHUB_REF: ${{ github.ref }}
+  GITHUB_EVENT_NAME: ${{ github.event_name }}
+
+jobs:
+  setup_environment:
+    runs-on: ubuntu-latest
+    outputs:
+      env_name: ${{ steps.get_env.outputs.env }}
+    steps:
+
+    - id: get_env
+      run: |
+        if grep -q "refs/tags/v" <<< $GITHUB_REF; then
+          echo "::set-output name=env::PROD"
+        else
+          echo "::set-output name=env::DEV"
+        fi
+
+    - id: print_env
+      name: Print environment
+      run: echo  "Environment :- ${{ steps.get_env.outputs.env }}"
+
+  build_and_deploy:
+    name: Build and deploy the go-clean-architecture repository
+    runs-on: ubuntu-latest
+    needs: setup_environment
+    environment: ${{needs.setup_environment.outputs.env_name}}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Use golang ${{matrix.go-version}}
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.18'
+      - run: go version
+      
+      - name: Set up MySQL
+        run: |
+          sudo systemctl start mysql
+          mysql -e 'CREATE DATABASE clean_architecture;' -uroot -proot
+          mysql -e "show databases;" -uroot -proot
+
+      - name: Initialize the environment variables for testing
+        run: |
+          echo "ENVIRONMENT=local
+          SERVER_PORT=8080
+          DB_HOST=localhost
+          DB_PORT=3306
+          DB_NAME=clean_architecture
+          DB_USER=root
+          DB_PASS=root
+          DEBUG_PORT=5002     
+          " > .env
+
+      - name: Initialize service account key file
+        run: |
+          echo '{
+            "type": "${{secrets.TYPE_SERVICE_ACCOUNT_KEY}}",
+            "project_id": "${{secrets.PROJECT_ID_SERVICE_ACCOUNT_KEY}}",
+            "private_key": "${{secrets.PRIVATE_KEY_SERVICE_ACCOUNT_KEY}}",
+            "private_key_id": "${{secrets.PRIVATE_KEY_ID_SERVICE_ACCOUNT_KEY}}",
+            "client_email": "${{secrets.CLIENT_EMAIL_SERVICE_ACCOUNT_KEY}}",
+            "client_id": "${{secrets.CLIENT_ID_SERVICE_ACCOUNT_KEY}}",
+            "auth_uri": "${{secrets.AUTH_URI_SERVICE_ACCOUNT_KEY}}",
+            "token_uri": "${{secrets.TOKEN_URI_SERVICE_ACCOUNT_KEY}}",
+            "auth_provider_x509_cert_url": "${{secrets.AUTH_PROVIDER_X509_CERT_URL_SERVICE_ACCOUNT_KEY}}",
+            "client_x509_cert_url": "${{secrets.CLIENT_X509_CERT_URL_SERVICE_ACCOUNT_KEY}}"
```

Comment:
@n3rdkid base64

Reply:
@dipeshdulal 
Let's not use it for now.